Framework/PythonInterface/plugins/algorithms/fractional_indexing.py

```diff
@@ -0,0 +1,302 @@
+import itertools
```

Comment:
Do we not have a space for helper classes at all?

Reply:
Not sure. I couldn't see an obvious place?

Reply:
Nor could I. I think other classes have done similar to this so I think it's fine to leave it